pkg/morph/deploy/alphabet.go

```diff
@@ -0,0 +1,100 @@
+	blockchain Blockchain
+
+	// based on blockchain
+	monitor *blockchainMonitor
+
+	committee keys.PublicKeys
+	localAcc  *wallet.Account
+}
+
+// initAlphabet designates NeoFS Alphabet role to all committee members on the
+// given Blockchain.
+func initAlphabet(ctx context.Context, prm initAlphabetPrm) error {
+	// wrap the parent context into the context of the current function so that
+	// transaction wait routines do not leak
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	committeeActor, err := newCommitteeNotaryActor(prm.blockchain, prm.localAcc, prm.committee)
+	if err != nil {
+		return fmt.Errorf("create Notary service client sending transactions to be signed by the committee: %w", err)
+	}
+
+	roleContract := rolemgmt.New(committeeActor)
+	txMonitor := newTransactionGroupMonitor(committeeActor)
+
+	for ; ; prm.monitor.waitForNextBlock(ctx) {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("wait for NeoFS Alphabet role to be designated for the committee: %w", ctx.Err())
+		default:
+		}
+
+		prm.logger.Info("checking NeoFS Alphabet role of the committee members...")
+
+		accsWithAlphabetRole, err := roleContract.GetDesignatedByRole(noderoles.NeoFSAlphabet, prm.monitor.currentHeight())
+		if err != nil {
+			prm.logger.Error("failed to check role of the committee, will try again later", zap.Error(err))
+			continue
+		}
+
+		someoneWithoutRole := len(accsWithAlphabetRole) < len(prm.committee)
+		if !someoneWithoutRole {
+			for i := range prm.committee {
+				if !accsWithAlphabetRole.Contains(prm.committee[i]) {
+					someoneWithoutRole = true
+					break
+				}
+			}
+		}
+		if !someoneWithoutRole {
+			prm.logger.Info("all committee members have a NeoFS Alphabet role")
+			return nil
+		}
+
+		prm.logger.Info("not all members of the committee have a NeoFS Alphabet role, designation is needed")
+
+		if txMonitor.isPending() {
+			prm.logger.Info("previously sent Notary request designating NeoFS Alphabet role to the committee is still pending, will wait for the outcome")
+			continue
+		}
+
+		tx, err := roleContract.DesignateAsRoleTransaction(noderoles.NeoFSAlphabet, prm.committee)
+		if err != nil {
+			prm.logger.Error("failed to make transaction designating NeoFS Alphabet role to the committee, will try again later", zap.Error(err))
+			continue
+		}
+
+		mainTxID, fallbackTxID, vub, err := committeeActor.Notarize(tx, nil)
+		if err != nil {
+			if isErrNotEnoughGAS(err) {
+				prm.logger.Info("insufficient Notary balance to send new Notary request designating NeoFS Alphabet role to the committee, skip")
+			} else {
+				prm.logger.Error("failed to send new Notary request designating NeoFS Alphabet role to the committee, skip", zap.Error(err))
+			}
+			continue
+		}
+
+		prm.logger.Info("Notary request designating NeoFS Alphabet role to the committee has been successfully sent, will wait for the outcome",
+			zap.Stringer("main tx", mainTxID), zap.Stringer("fallback tx", fallbackTxID), zap.Uint32("vub", vub))
```

Comment:
```suggestion
			zap.Stringer("main_tx", mainTxID), zap.Stringer("fallback_tx", fallbackTxID), zap.Uint32("vub", vub))
```

suggest not to use spaces in json format. i think we do not have them in our code at all

Reply:
json doesn't forbid whitespaces, so what's the problem?

Reply:
create an issue, this aint a problem to me here cuz JSON specification doesn't forbid whitespaces. If we'll decide to get rid of them in log messages, this should be done globally

Reply:
https://github.com/nspcc-dev/neofs-node/issues/2582